app/ui/src/app/integration/edit-page/current-flow.service.ts

```diff
@@ -716,43 +695,78 @@ export class CurrentFlowService {
   ): any {
     switch (step.stepKind) {
       case SPLIT:
-      case AGGREGATE:
         // A split step needs the data shape of the previous thing with a data shape
         const prev = this.getPreviousStepWithDataShape(position);
+        this.fetchStepDescriptor(
+          step,
+          prev.action.descriptor.inputDataShape,
+          prev.action.descriptor.outputDataShape,
+          then
+        );
+        break;
+      case AGGREGATE:
+        // An aggregate step needs the data shape of the subsequent thing with a data shape
         const subsequent = this.getSubsequentStepWithDataShape(position);
-        const subsequentDataShape =
-          subsequent !== undefined
-            ? subsequent.action.descriptor.inputDataShape
-            : undefined;
-        const sub = this.integrationSupportService
-          .getStepDescriptor(step.stepKind, {
-            inputShape:
-              step.stepKind === AGGREGATE
-                ? subsequentDataShape
-                : prev.action.descriptor.inputDataShape,
-            outputShape: prev.action.descriptor.outputDataShape,
-          })
-          .subscribe(
-            descriptor => {
-              step = {
-                ...step,
-                action: { actionType: 'step', descriptor } as Action,
-              };
-              sub.unsubscribe();
-              then(step);
-            },
-            err => {
-              // we'll just pass through
-              sub.unsubscribe();
-              then(step);
-            }
-          );
+        this.fetchStepDescriptor(
+          step,
+          subsequent.action.descriptor.inputDataShape,
+          subsequent.action.descriptor.outputDataShape,
```

Comment:
Aggregate should be provided with the outputDataShape of the previous step and inputDataShape of the subsequent step

Reply:
Perfect, thanks will update today!